content/applications/productivity/whatsapp.rst

```diff
@@ -7,17 +7,665 @@ content. Businesses can use `WhatsApp Business
+   administrator of the Meta account when the template has been approved.
+
+Using placeholders and variables
+--------------------------------
+
+Dynamic variables reference certain fields within the Odoo database to produce unique data in the
+WhatsApp message when using a template. Dynamic variables are encoded to display fields from within
+the database referencing fields from within a model.
+
+.. example::
+   Many companies like to customize their WhatsApp messages with a personalized piece of customer
+   information to grab attention. This can be accomplished in Odoo by referencing a field within a
+   model by settings a dynamic variable. For example, a customer's name can be referenced in the
+   email from the :guilabel:`Customer` field on the :guilabel:`Sales Order` model.
+
+Dynamic variables can be added in to the :guilabel:`Body` by adding :guilabel:`placeholders` in the
+*text*. To add a placeholder in the *message body* enter the following text `{{1}}`. For the second
+placeholder enter `{{2}}` and  increase incrementally as more placeholders are added to the text.
+
+.. example::
+   Hello {{1}},
+
+   Use the following link to pay *{{2}}{{3}}* to *{{4}}* with reference *{{5}}*.
+   {{6}}
+
+   Don't share screenshots of payment here.
+
+   Thank you
+
+.. seealso::
+   :ref:`whatsapp/templates`
+
+These placeholders must be configured on the :guilabel:`Variables` tab of the template before
+submitting for approval from Meta. To edit the dynamic variables on a template, first change the
+:guilabel:`Type` to :guilabel:`Field of Model`. This allows Odoo to reference a field within a model
+to produce unique data in the message being sent.
+
+Next, go to the :guilabel:`Variables` tab and edit the :guilabel:`Field` field of the dynamic
+variables. The :guilabel:`Applies to` field in the template can be edited prior to ensure the
+correct model and field are referenced.
+
+To search the available fields, type in the front-end name (on user-interface) of the field in the
+search box. This will find a result from all of the available fields for the model
+(:guilabel:`Applies to`) that the template is created for. There may be multiple levels that need to
+be configured.
+
+.. example::
+   The following is an example of the variables set for the above placeholders in a payment
+   template:
+
+   .. image:: whatsapp/set-variables.png
+      :align: center
+      :alt: An example of the variables set for the placeholders set in a payment template.
+
+.. warning::
+   Customizing WhatsApp templates is out of the scope of Odoo Support.
+
+After updating the dynamic variables on the template, the template needs to be submitted to Meta for
+approval again. Click :guilabel:`Submit for Approval` to start the approval process. An email will
+be sent to the administrator of the Meta account when the template has been approved.
+
+Following the approval from Meta, sync the templates again in the Odoo database. See this
```

Comment:
Yu probably don't know, but are the pre-configured templates "pre-approved" from Meta in any way? I just wondered how long this process typically is, and if it is faster if using the pre-configured ones. Not sure if support would know this or not. Not necessary info but if it's available it would be good to add. (Like does Meta have an approval pending dashboard or something we can reference? Just so we can point user's to see the status and have an idea about the process))

Reply:
Added link to Meta portal